tasks/main.yml

```diff
@@ -145,9 +145,31 @@
   command: "{{ puppet_agent_bin_path }} agent --enable"
   when: puppetize_enable_puppet
 
+- name: Dry run (no changes, depending on the puppet code) of Puppet
+  command: "{{ puppet_agent_bin_path }} agent -t --noop"
+  register: dry_puppet_result
+  changed_when: dry_puppet_result.rc == 2
+  failed_when: (dry_puppet_result.rc != 2) and (dry_puppet_result.rc != 0)
+  tags: [ 'puppet_run', 'skip_ansible_lint' ]
+  when: noop_run
+
+- name: Show output
+  debug:
+    var: dry_puppet_result['stdout_lines']
+  when: noop_run
+
 - name: puppetize!
-  command: "{{ puppet_agent_bin_path }} agent -t"
+  command: "{{ puppet_agent_bin_path }} agent -t --logdest /tmp/puppetize"
```

Comment:
This logs to a specific log file, but not when running a noop run?

Reply:
I left it by mistake from my test. Removed now.